### ISSUE TITLE
Transaction Calculations incorrect

### ISSUE THREAD

**Issue by Napier** [iConomy/Core#16]
Did some testing on my test server, and got some wonky results with pay/grant commands.

Using: cb493, iCo 4.4.3 with MySQL and Cache=true, Permissions 2.5.2

Using /money grant command:
1. log on with a starting balances of 50
2. use command /money grant playername 100
    Console output: "Player 250's account had 100.00 grant to it."
    Ingame output: "100 coin was credited to your account...Balance 250 coin...<PlayerName>'s account had 100 coin credited."
3  Use command /money --> and the resulting amount it displays is 250 coins.
4  I log off, reload server, log back on, and it the account is still at 250 coins.

Using /money pay command:
1. 2 players start with a balances of: Player 1 - 1203.74 / Player 2 - 0.17
2. Player 1 does a /money pay player2 100
3. Results: Player 1 ends up with 11.04 after the transaction. And player 2 ends up with 200.17.
4. Reload server, and it the accounts values remain at the values in #3.

However, with Cache=false:
- The grant command seems to work as expected.
- But the Pay command still produces odd results similar to the above.

**Comment by nijikokun** [comment]
Could you write this a little better? It's hard to follow, as I cannot reproduce this at all.

Cache: true, is there for a reason; It updates the database when a change has occured if told to.
Cache: false, updates every single time.

Regardless of amount, and the 11.04 I cannot reproduce that.

**Comment by Napier** [comment]
With regard to the /money pay command, the amounts were not being decremented as I would have expected.

Using this scenario:
Step 1. 2 players start with a balances of: Player1 - 1203.74 / Player2 - 0.17
Step 2. Player1 does a /money pay Player2 100

Expected results were to have the subsequent account balances be:
Player1 - 1103.74
Player2 - 100.17

But instead, the account balances ended at:
Player1 - 11.04
Player2 - 200.17

Catox in the thread mentions that there was a division action being used in the process of decrementing the amount of 100 from Player1's account, and that seems to be what was happening in the above scenario. eg. the starting balance was 1203.74, then 100 was subtracted from it to equal - 1103.74, but then an additional division operation was performed - 1103.74 / 100 = 11.0374 (rounded up to 11.04).

**Comment by crohakon** [comment]
We are having this same issue on our server as well.

**Comment by nijikokun** [comment]
Has this been fixed?

**Comment by Napier** [comment]
It appears to be yes. I have not encountered an issue since v4.4.5.  Thanks for the support!
